src/Avalonia.Controls/SelectionModel.cs

```diff
@@ -36,9 +36,12 @@ public SelectionModel()
             get => _rootNode?.Source;
             set
             {
-                ClearSelection(resetAnchor: true, raiseSelectionChanged: false);
+                using (var operation = new Operation(this))
```

Comment:
Would it make sense to expose this ? That way if an app is making multiple changes, they could batch all of them and get a single collection changed event ?

Reply:
Very good point!